### PR TITLE
Initial Differential Expression of Northcott-Processed Data (Validate DE Part 1)

### PR DESCRIPTION
**Pull request by cansav09**
This is part 1 of probably 2. This is just the initial notebook that covers steps 1 and 2 outlined by @jaclyn-taroni in issue #28. 

Part 2 will be Steps 3-5.

There is also file reorganization to get rid of the master data file and have each section have it's own data file. This is to address issue #42

### CODE REVIEW

File: clustering/clustering_example.Rmd [hunk 1]
```diff
@@ -96,7 +96,7 @@ and filename within the function to the respective file suffix.
 
 ```{r Save heatmap as a png}
 # Open a png file
-png("HeatmapGSE12955.png")
+png(file.path("results", "HeatmapGSE12955.png"))
```
Comment (jaclyn-taroni) [inline clustering/clustering_example.Rmd]: This should be in a `plots` directory that should be created in this notebook.

Comment (jaclyn-taroni) [reply clustering/clustering_example.Rmd]: The original file is also still in the top directory of this module as far as I can tell.

File: validate-differential-expression/author_processed_DE.Rmd [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+stats <- topTable(fit, number = nrow(author.df))
+
+# Print out stats
+stats
+```
+
+## 6) Write statistics to output results file
+
+```{r  Write results}
+readr::write_tsv(stats, file.path("results", "GSE37382_limma_results.tsv"))
```
Comment (jaclyn-taroni) [inline validate-differential-expression/author_processed_DE.Rmd]: I don't see this file here.

File: validate-differential-expression/author_processed_DE.Rmd [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+## 2) Import and set up author processed data from GEO
+For more information on how this data was processed, see its 
+[GEO entry](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37382)
+
+```{r}
+# Store file name for gene matrix
+matrix.file.name <- file.path("data", "GSE37382_series_matrix.txt")
+
+# Download author processed data file directly to `data` folder
+geo.data <-  GEOquery::getGEO("GSE37382", destdir = "data", GSEMatrix = TRUE)
```
Comment (jaclyn-taroni) [inline validate-differential-expression/author_processed_DE.Rmd]: I think this all may be in the wrong spot? Based on our in person discussion, I suspect you did not move all the files to the new `validate-differential-expression` module.

File: README.md [hunk 4]
```diff
@@ -83,8 +83,14 @@ file from your data, you can create a a CLS format using GenePattern's online
...
 2. [Annotation using Ensembl IDs](https://github.com/AlexsLemonade/refinebio-examples/blob/master/ensembl-id-convert)  
 
-3. [Differential Expression Analyses](https://github.com/AlexsLemonade/refinebio-examples/blob/master/differential-expression)  
+3. [Differential expression analyses](https://github.com/AlexsLemonade/refinebio-examples/blob/master/differential-expression)  
 
-4. [Ortholog Mapping](https://github.com/AlexsLemonade/refinebio-examples/blob/master/ortholog-mapping)  
+4. [Ortholog mapping](https://github.com/AlexsLemonade/refinebio-examples/blob/master/ortholog-mapping)  
 
-5. [Other scripts](https://github.com/AlexsLemonade/refinebio-examples/blob/master/scripts)  
+5. [Batch correction with ComBat](https://github.com/AlexsLemonade/refinebio-examples/blob/master/batch_correction)
```
Comment (jaclyn-taroni) [inline README.md]: This should be taken out until the example gets merged in.

**Comment by cansav09** [comment]
Okay. Thanks for the review. Sorry for the messiness of this PR. Ready for you to look again.

File: validate-differential-expression/author_processed_DE.Rmd [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+## 6) Write statistics to output results file
+
+```{r  Write results}
+# Save summary statistics to a tsv file
+readr::write_tsv(stats, file.path("results", "GSE37382_limma_results.tsv"))
+
+# Save the limma fit object to an RDS for later use
+saveRDS(fit2, file.path("results", "GSE37382_limma_fit_obj.RDS"))
```
[the remark this answers is not in the document]
Comment (cansav09) [reply validate-differential-expression/author_processed_DE.Rmd]: I thought I might be using it for making the VennDiagram but decided against it for generalizability purposes.

File: differential-expression/gene_DE.Rmd [hunk 6]
```diff
@@ -95,17 +109,36 @@ gene expression data can be found [here](http://www.nathalievialaneix.eu/doc/pdf
...
+
+# Print out the plot here
+voc.plot
 ```
 
 ## 5) Write statistics to output results file
 
 ```{r  Write results}
-write.table(stats, file = "GSE71270limmaResults.tsv", sep = "\t", quote = FALSE)
+readr::write_tsv(stats, file.path("results", "GSE71270_limma_results.tsv"))
```
[the remark this answers is not in the document]
Comment (cansav09) [reply differential-expression/gene_DE.Rmd]: Added a `tibble::rownames_to_column` to fix that.

File: validate-differential-expression/author_processed_DE.Rmd [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+## 3) Extract metadata for this dataset
+
+```{r}
+# Extract metadata
+metadata <- data.frame(geo.data[[1]]@phenoData@data) %>% 
+  # Get rid of columns that have only NAs
+  dplyr::select(-which(apply(is.na(.), 2, all))) %>% 
+  readr::write_tsv(file.path("data", "geo_obtained_GSE37382_metadata.tsv"))
```
[the remark this answers is not in the document]
Comment (cansav09) [reply validate-differential-expression/author_processed_DE.Rmd]: No, I don't intend to track it. It's not really needed for other notebooks, and the other GEO files are obtained straight from GEO. It's more for the users' own purposes I guess. I could delete it if we want.

File: README.md [hunk 8]
```diff
@@ -83,8 +83,12 @@ file from your data, you can create a a CLS format using GenePattern's online
...
-3. [Differential Expression Analyses](https://github.com/AlexsLemonade/refinebio-examples/blob/master/differential-expression)  
+3. [Differential expression analyses](https://github.com/AlexsLemonade/refinebio-examples/blob/master/differential-expression)  
 
-4. [Ortholog Mapping](https://github.com/AlexsLemonade/refinebio-examples/blob/master/ortholog-mapping)  
+4. [Ortholog mapping](https://github.com/AlexsLemonade/refinebio-examples/blob/master/ortholog-mapping)  
 
-5. [Other scripts](https://github.com/AlexsLemonade/refinebio-examples/blob/master/scripts)  
+5. [Quantile normalizing your own data](https://github.com/AlexsLemonade/refinebio-examples/blob/master/normalize-own-data)
+
+6. [Validate differential expression](https://github.com/AlexsLemonade/refinebio-examples/blob/master/validate-differential-expression)
```
[the remark this answers is not in the document]
Comment (cansav09) [reply README.md]: Sure.

File: validate-differential-expression/author_processed_DE.Rmd (hunk 7 above)
[the remark this answers is not in the document]
Comment (cansav09) [reply validate-differential-expression/author_processed_DE.Rmd]: I got rid of it. It's not really necessary.

**Pull request merged**